screengrab/lib/screengrab/runner.rb

```diff
@@ -210,27 +220,41 @@ def grant_permissions(device_serial)
       end
     end
 
-    def run_tests(device_serial, test_classes_to_use, test_packages_to_use, launch_arguments)
+    def run_tests(device_serial, app_apk_path, tests_apk_path, test_classes_to_use, test_packages_to_use, launch_arguments)
+      unless @config[:reinstall_app]
+        install_apks(device_serial, app_apk_path, tests_apk_path)
+        grant_permissions(device_serial)
+      end
+
       @config[:locales].each do |locale|
-        UI.message "Running tests for locale: #{locale}"
-
-        instrument_command = ["adb -s #{device_serial} shell am instrument --no-window-animation -w",
-                              "-e testLocale #{locale.tr('-', '_')}",
-                              "-e endingLocale #{@config[:ending_locale].tr('-', '_')}"]
-        instrument_command << "-e class #{test_classes_to_use.join(',')}" if test_classes_to_use
-        instrument_command << "-e package #{test_packages_to_use.join(',')}" if test_packages_to_use
-        instrument_command << launch_arguments.map { |item| '-e ' + item }.join(' ') if launch_arguments
-        instrument_command << "#{@config[:tests_package_name]}/#{@config[:test_instrumentation_runner]}"
-
-        test_output = run_adb_command(instrument_command.join(" \\\n"),
-                                      print_all: true,
-                                      print_command: true)
-
-        if @config[:exit_on_test_failure]
-          UI.user_error!("Tests failed", show_github_issues: false) if test_output.include?("FAILURES!!!")
-        else
-          UI.error("Tests failed") if test_output.include?("FAILURES!!!")
+        if @config[:reinstall_app]
+          uninstall_apks(device_serial, @config[:app_package_name], @config[:tests_package_name])
+          install_apks(device_serial, app_apk_path, tests_apk_path)
+          grant_permissions(device_serial)
```

Comment:
could this get bit simplified?

like trying to get rid of the `unless` above?

could this work (and removing the unless + the if here?):

```ruby
uninstall_apks(device_serial, @config[:app_package_name], @config[:tests_package_name]) if @config[:reinstall_app]

```